### PR TITLE
Feat/kube restrictions

### PR DESCRIPTION
**Pull request by marie-j**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         |no
| New feature?     | yes
| Breaking change? |no
| Tickets          | MANAGER-5296
| License          | BSD 3-Clause

## Related
- [x] TRANS-31973

### CODE REVIEW

**Review by mohammed-zahaf — APPROVED** [review]

**Review by antleblanc — APPROVED** [review]

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/restrictions/IP.class.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export default class IP {
```
Comment (JeremyDec) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/restrictions/IP.class.js:1]: **nitpick**:
Regarding https://github.com/ovh/manager/pull/4316/files#diff-182ca3baad5b54648cbcbcd1b8ebdb575febbba4a6bdbc25f165f8fe9d2f09e3R75, should the name of the class be `IPv4` ?

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/restrictions/controller.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+          `${this.CucCloudMessage.error(
+            this.$translate.instant('kube_restrictions_add_error'),
+          )}${error.data}`,
+      )
+      .finally(() => {
+        this.isLoading = false;
+      });
+  }
+
+  isIPValid() {
```
Comment (JeremyDec) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/restrictions/controller.js:95]: Why not having this method into the `IP` class ?

Comment (marie-j) [reply packages/manager/modules/pci/src/projects/project/kubernetes/details/restrictions/controller.js:95]: I tried but if I use an array value such as `this.ips[$rowIndex]`, the input looses focus after each character (probably due to the way ngrepeat is handled by the datagrid tracking by $index https://github.com/angular/angular.js/issues/13327) so I have to workaround by using a different value which is not an IP instance

Comment (JeremyDec) [reply packages/manager/modules/pci/src/projects/project/kubernetes/details/restrictions/controller.js:95]: Ok good to know

**Comment by marie-j** [comment]
Only change made since review was  to rename IP class to IPv4

**Review by JeremyDec — DISMISSED** [review]

**Review by antleblanc — DISMISSED** [review]

**Review by antleblanc — APPROVED** [review]

**Review by JeremyDec — APPROVED** [review]

**Pull request merged**